### PR TITLE
fixup! gnrc_ipv6: fix asserts for NHC

### PR DESCRIPTION
**Pull request by Yonezawa-T2**
PR for https://github.com/RIOT-OS/RIOT/pull/5281#issuecomment-211822054

### CODE REVIEW

File: sys/include/net/gnrc/netapi.h [hunk 1]
```diff
@@ -91,11 +91,12 @@ int gnrc_netapi_send(kernel_pid_t pid, gnrc_pktsnip_t *pkt);
  * @param[in] demux_ctx demultiplexing context for @p type.
  * @param[in] cmd       command for all subscribers
  * @param[in] pkt       pointer into the packet buffer holding the data to send
+ * @param[in] release   releases the pkt after dispatching if true
  *
  * @return Number of subscribers to (@p type, @p demux_ctx).
  */
 int gnrc_netapi_dispatch(gnrc_nettype_t type, uint32_t demux_ctx, uint16_t cmd,
-                         gnrc_pktsnip_t *pkt);
+                         gnrc_pktsnip_t *pkt, bool release);
```
Comment (authmillenon) [inline sys/include/net/gnrc/netapi.h]: The problem with this extra parameter is that at least the ARM cortex platforms will now start to put the parameters on the stack and increase the ROM size for every call to `gnrc_netapi_dispatch*()` now. So I'd rather keep the wrapping function in IPv6 for the special use case in that module.

Comment (Yonezawa-T2) [inline sys/include/net/gnrc/netapi.h]: Fixed.

Should we also change `gnrc_netapi_dispatch_receive` and `gnrc_netapi_dispatch_send` to non-inline functions?

Comment (authmillenon) [inline sys/include/net/gnrc/netapi.h]: Nahh, I take it as is.

File: sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c [hunk 2]
```diff
@@ -177,9 +135,80 @@ void gnrc_ipv6_ext_demux(kernel_pid_t iface,
...
+                break;
+
+            default:
+                gnrc_ipv6_demux(iface, current, pkt, nh); /* demultiplex next header */
+                return;
+        }
+    }
 
-    return;
+    assert(false); /* never reaches here */
```
Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c]: Then why have the assert here?

File: sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c [hunk 3]
```diff
@@ -177,9 +135,80 @@ void gnrc_ipv6_ext_demux(kernel_pid_t iface,
...
+                nh = ext->nh;
+                DEBUG("ipv6_ext: next header = %" PRIu8 "\n", nh);
+
+                if ((current = _mark_extension_header(current, &pkt)) == NULL) {
+                    return;
+                }
+
+                gnrc_netapi_dispatch(GNRC_NETTYPE_IPV6, nh, GNRC_NETAPI_MSG_TYPE_RCV, pkt, false);
+
+                break;
```
Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c]: This break and above contradict the comment below.

Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c]: Args sorry, did not see the loop.

File: sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c [hunk 4]
```diff
@@ -139,74 +180,40 @@ void gnrc_ipv6_demux(kernel_pid_t iface, gnrc_pktsnip_t *current, gnrc_pktsnip_t
...
-    else {
-        if (!interested) {
-#ifdef MODULE_GNRC_SIXLOWPAN_IPHC_NHC
-            /* second statement is true for small 6LoWPAN NHC decompressed frames
-             * since in this case it looks like
-             *
-             * * GNRC_NETTYPE_UDP <- pkt    (or GNRC_NETTYPE_IPV6_EXT respectively)
-             * v
-             * * GNRC_NETTYPE_UDP <- current
+             * * GNRC_NETTYPE_EXT
```
Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c]: NHC does not support extension headers yet, so I'm not sure this comment should be as is.

Comment (Yonezawa-T2) [inline sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c]: It should be supported in the future. So I prefer to mention it here.

Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c]: okay.

**Comment by authmillenon** [comment]
Except for https://github.com/authmillenon/RIOT/pull/28#discussion_r60368548 I would merge now.

File: sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c (hunk 2 above)
Comment (Yonezawa-T2) [inline sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c]: We will return from `case PROTNUM_IPV6_EXT_RH:` or `default:`. So we will never reach here.

Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c]: Wouldn't be my style, but alright, I take it.

**Pull request merged**